File: apps/api/app/schemas/agent_role_status.py
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.common import FixtureEnvelope

AgentRoleState = Literal["대기", "승인 필요", "실패 이력"]


class AgentRoleStatusItem(BaseModel):
    model_config = ConfigDict(extra="forbid")

    id: str
    role: str
    status: AgentRoleState
    badge: str
    task: str
    wait: str
    approval: bool
    history: str
    decision: str
    summary: str
    conflict: str
    linkPage: str
    linkLabel: str


class AgentRoleStatusData(BaseModel):
    model_config = ConfigDict(extra="forbid")

    title: str
    safetyCopy: str
    roles: list[AgentRoleStatusItem] = Field(min_length=1)


class AgentRoleStatusEnvelope(FixtureEnvelope):
    data: AgentRoleStatusData

File: apps/web/src/types/agentRoleStatus.ts
import type { components } from "./api.generated";
import type { Api, FixtureEnvelope } from "./common";

export type AgentRoleState = "대기" | "승인 필요" | "실패 이력";

export type AgentRoleStatusItem = Api<components["schemas"]["AgentRoleStatusItem"]>;
export type AgentRoleStatusData = Api<components["schemas"]["AgentRoleStatusData"]>;
export type AgentRoleStatusEnvelope = FixtureEnvelope<AgentRoleStatusData>;
